broker-daemon/utils/retry.spec.js

```diff
@@ -0,0 +1,88 @@
+const { sinon, rewire, expect } = require('test/test-helper')
+const path = require('path')
+
+const retry = rewire(path.resolve(__dirname, 'retry'))
+
+describe('retry', () => {
+  let callFunction
+  let message
+  let attempts
+  let delayTime
+  let logger
+  let delay
+  let res
+
+  beforeEach(() => {
+    res = 'response'
+    callFunction = sinon.stub().resolves(res)
+    message = 'test message'
+    attempts = 2
+    delayTime = 100
+    logger = { error: sinon.stub() }
+    delay = sinon.stub().resolves()
+    retry.__set__('logger', logger)
+    retry.__set__('delay', delay)
+  })
+
+  it('calls the call function', async () => {
+    await retry(callFunction, message, attempts, delayTime)
+
+    expect(callFunction).to.be.to.have.been.calledOnce()
+  })
+
+  it('returns the response of the callFunction if there were no errors', async () => {
```

Comment:
Probably worth it to make sure that `callFunction` is called only once if it has no errors on the first time